tests/Util/RequestParserTest.php

```diff
@@ -31,5 +30,17 @@ public function testParseRequestParams()
     {
         $value = RequestParser::parseRequestParams('gerard.name=dargent');
         $this->assertEquals($value, ['gerard.name' => 'dargent']);
+
+        // urlencoded + (plus) in query string.
+        $value = RequestParser::parseRequestParams('date=2000-01-01T00%3A00%3A00%2B00%3A00');
```

Comment:
We might want to use a data provider... https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers